ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -513,13 +524,32 @@ def process_cluster_cli_params(config):
     skip_download_client = get_cli_param(config, "skip_download_client")
     if skip_download_client:
         ocsci_config.DEPLOYMENT["skip_download_client"] = True
+    re_trigger_failed_tests = get_cli_param(config, "--re-trigger-failed-tests")
+    if re_trigger_failed_tests:
+        ocsci_config.RUN["re_trigger_failed_tests"] = os.path.expanduser(
+            re_trigger_failed_tests
+        )
 
 
 def pytest_collection_modifyitems(session, config, items):
     """
     Add Polarion ID property to test cases that are marked with one.
     """
-    for item in items:
+
+    re_trigger_failed_tests = ocsci_config.RUN.get("re_trigger_failed_tests")
+    if re_trigger_failed_tests:
+        junit_report = JUnitXml.fromfile(re_trigger_failed_tests)
+        cases_to_re_trigger = []
+        for suite in junit_report:
+            cases_to_re_trigger += [_case.name for _case in suite if _case.result]
```

Comment:
anything other than pass will be executed based on if condition? (that is it includes skipped)

Reply:
We want to include skipped tests as if cluster is not healthy because some test broke the cluster the rest of the tests are skipped, those we would like to trigger again. If the test case should be skipped because some skipif in other code condition, those will be still skipped also in the new execution.